diff --git a/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces b/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
index 063ab1a1..9f847082 100755
--- a/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
+++ b/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
@@ -71,6 +71,11 @@ elseif platform.match('lantiq', 'xrx200', {
 	'arcadyan,vgv7510kw22-nor',
 }) then
 	lan_ifname, wan_ifname = 'lan1 lan2 lan3 lan4', 'wan'
+elseif platform.match('lantiq', 'xrx200', {
+	'tplink,tdw8970',
+	'tplink,tdw8980',
+}) then
+	lan_ifname, wan_ifname = 'lan1 lan2 lan3', 'lan4'
 elseif platform.match('realtek', 'rtl838x', {
 	'd-link,dgs-1210-10p',
 }) then
