diff --git a/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces b/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
index 5e5243a0..81ff6513 100755
--- a/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
+++ b/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
@@ -79,6 +79,21 @@ elseif platform.match('lantiq', 'xrx200', {
 	'arcadyan,vgv7510kw22-nor',
 }) then
 	lan_ifname, wan_ifname = 'lan1 lan2 lan3 lan4', 'wan'
+elseif platform.match('lantiq', 'xrx200', {
+	'avm,fritz3370-rev2-hynix',
+	'avm,fritz3370-rev2-micron',
+	'avm,fritz3390',
+	'avm,fritz7360sl',
+	'avm,fritz7360-v2',
+	'avm,fritz7362sl',
+	'avm,fritz7430',
+}) then
+	lan_ifname, wan_ifname = 'lan2 lan3 lan4', 'lan1'
+elseif platform.match('lantiq', 'xrx200', {
+	'tplink,tdw8970',
+	'tplink,tdw8980',
+}) then
+	lan_ifname, wan_ifname = 'lan1 lan2 lan3', 'lan4'
 elseif platform.match('realtek', 'rtl838x', {
 	'd-link,dgs-1210-10p',
 }) then
