diff --git a/docs/user/supported_devices.rst b/docs/user/supported_devices.rst
index 71d85509..9ee2ab7e 100644
--- a/docs/user/supported_devices.rst
+++ b/docs/user/supported_devices.rst
@@ -150,6 +150,10 @@ ath79-generic
 
 * Ubiquiti
 
+  - NanoBeam 5AC 19 (XC)
+  - NanoBeam M5 (XW)
+  - NanoStation Loco M2/M5 (XW)
+  - NanoStation M2/M5 (XW)
   - UniFi AC Lite
   - UniFi AC LR
   - UniFi AC Mesh
diff --git a/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces b/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
index 98c97ed9..e6339261 100755
--- a/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
+++ b/package/gluon-core/luasrc/lib/gluon/upgrade/020-interfaces
@@ -50,6 +50,7 @@ if platform.match('ath79', 'generic', {
 	'tplink,wbs210-v1',
 	'tplink,wbs210-v2',
 	'tplink,wbs510-v1',
+	'ubnt,nanostation-m-xw',
 	'ubnt,unifi-ap-pro',
 }) then
 	lan_ifname, wan_ifname = wan_ifname, lan_ifname
diff --git a/package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua b/package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua
index f3ea1c93..8bc3e767 100644
--- a/package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua
+++ b/package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua
@@ -42,6 +42,10 @@ function M.is_outdoor_device()
 		'tplink,wbs210-v1',
 		'tplink,wbs210-v2',
 		'tplink,wbs510-v1',
+		'ubnt,nanobeam-ac-xc',
+		'ubnt,nanobeam-m5-xw',
+		'ubnt,nanostation-loco-m-xw',
+		'ubnt,nanostation-m-xw',
 		'ubnt,uk-ultra',
 		'ubnt,unifi-ap-outdoor-plus',
 		'ubnt,unifiac-mesh',
diff --git a/targets/ath79-generic b/targets/ath79-generic
index 0e317056..aeb30bc2 100644
--- a/targets/ath79-generic
+++ b/targets/ath79-generic
@@ -529,6 +529,18 @@ device('tp-link-wbs510-v1', 'tplink_wbs510-v1')
 
 -- Ubiquiti
 
+device('ubiquiti-nanobeam-ac-gen1-xc', 'ubnt_nanobeam-ac-xc', {
+	packages = ATH10K_PACKAGES_QCA9880,
+})
+
+device('ubiquiti-nanostation-loco-m-xw', 'ubnt_nanostation-loco-m-xw', {
+	aliases = {
+		'ubiquiti-nanobeam-m5-xw',
+	},
+})
+
+device('ubiquiti-nanostation-m-xw', 'ubnt_nanostation-m-xw')
+
 device('ubiquiti-unifi-ac-lite', 'ubnt_unifiac-lite', {
 	factory = false,
 	packages = ATH10K_PACKAGES_QCA9880,
