diff --git a/targets/ramips-mt7621 b/targets/ramips-mt7621
index 3a2c1cd2..630fd7ab 100644
--- a/targets/ramips-mt7621
+++ b/targets/ramips-mt7621
@@ -37,6 +37,12 @@ device('cudy-x6-v2', 'cudy_x6-v2', {
 
 -- D-Link
 
+device('d-link-covr-x1860-a1', 'dlink_covr-x1860-a1', {
+	extra_images = {
+		{'-squashfs-recovery', '-recovery', '.bin'}
+	},
+})
+
 device('d-link-dap-x1860-a1', 'dlink_dap-x1860-a1')
 
 device('d-link-dir-860l-b1', 'dlink_dir-860l-b1')
