diff --git a/package/gluon-core/luasrc/lib/gluon/upgrade/250-cellular b/package/gluon-core/luasrc/lib/gluon/upgrade/250-cellular
index 50eabdc6..3c59f750 100755
--- a/package/gluon-core/luasrc/lib/gluon/upgrade/250-cellular
+++ b/package/gluon-core/luasrc/lib/gluon/upgrade/250-cellular
@@ -57,6 +57,7 @@ elseif platform.match('ath79', 'nand', {
 elseif platform.match('ipq40xx', 'generic', {
 	'glinet,gl-ap1300',
 	'zte,mf289f',
+	'zte,mf286r',
 }) then
 	setup_ncm_qmi('/dev/cdc-wdm0', 'qmi', 15)
 elseif platform.match('ramips', 'mt7621', {
diff --git a/package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua b/package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua
index 163e8e41..66604ffd 100644
--- a/package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua
+++ b/package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua
@@ -91,6 +91,7 @@ function M.is_cellular_device()
 	elseif M.match('ipq40xx', 'generic', {
 		'glinet,gl-ap1300',
 		'zte,mf289f',
+		'zte,mf286r',
 	}) then
 		return true
 	elseif M.match('ramips', 'mt7621', {
