diff --git a/modules b/modules
index 8b77b64d..c0da2c9a 100644
--- a/modules
+++ b/modules
@@ -1,8 +1,8 @@
 GLUON_FEEDS='gluon packages routing'
 
-OPENWRT_REPO=https://github.com/openwrt/openwrt.git
-OPENWRT_BRANCH=openwrt-23.05
-OPENWRT_COMMIT=59fd8f08cf544a7f0fcaf700ca9af9292de3adcc
+OPENWRT_REPO=https://github.com/maurerle/openwrt.git
+OPENWRT_BRANCH=covr-x1860
+OPENWRT_COMMIT=59ade364e3de2a3353e7050004d4165dd48c3ab1
 
 PACKAGES_GLUON_REPO=https://github.com/freifunk-gluon/packages.git
 PACKAGES_GLUON_COMMIT=53ea3b89771fc7d7a80f1800ce25e98dfe1633aa
