diff --git a/modules b/modules
index 1d7e2baa..1699a21c 100644
--- a/modules
+++ b/modules
@@ -2,7 +2,7 @@ GLUON_FEEDS='gluon packages routing'
 
 OPENWRT_REPO=https://github.com/openwrt/openwrt.git
 OPENWRT_BRANCH=openwrt-23.05
-OPENWRT_COMMIT=0cdbbd8868030f7b1dc0e7f9fcff58bb3a702adf
+OPENWRT_COMMIT=50148a40d2b0de8adddd8f11b95d6436355c431e
 
 PACKAGES_GLUON_REPO=https://github.com/freifunk-gluon/packages.git
 PACKAGES_GLUON_COMMIT=3d08b0fee8dc5d96d8bcdb985fad1d5564de4022
