diff --git a/targets/ramips-mt7621 b/targets/ramips-mt7621
index 01b9b94c..630fd7ab 100644
--- a/targets/ramips-mt7621
+++ b/targets/ramips-mt7621
@@ -22,6 +22,10 @@ device('cudy-wr2100', 'cudy_wr2100', {
 	factory = false,
 })
 
+device('cudy-m1800', 'cudy_m1800', {
+	factory = false,
+})
+
 device('cudy-x6-v1', 'cudy_x6-v1', {
 	factory = false,
 })
@@ -89,6 +93,8 @@ device('netgear-wndr3700-v5', 'netgear_wndr3700-v5', {
 
 -- TP-Link
 
+device('tp-link-archer-ax23-v1', 'tplink_archer-ax23-v1')
+
 device('tp-link-archer-c6-v3', 'tplink_archer-c6-v3', {
 	broken = true, -- LAN LED not working - review after resolving #2756
 })
@@ -106,6 +112,10 @@ device('ubiquiti-unifi-6-lite', 'ubnt_unifi-6-lite', {
 	factory = false,
 })
 
+device('ubiquiti-unifi-nanohd', 'ubnt_unifi-nanohd', {
+	factory = false,
+})
+
 
 -- Wavlink
 
@@ -162,6 +172,15 @@ device('zyxel-wsm20', 'zyxel_wsm20', {
 
 -- Devices without WLAN
 
+-- Mikrotik
+
+device('mikrotik-routerboard-750gr3', 'mikrotik_routerboard-750gr3', {
+	aliases = {'mikrotik-routerboard-hex-v3'},
+	factory = false,
+	packages = {'-hostapd-mini'},
+})
+
+
 -- Ubiquiti
 
 device('ubiquiti-edgerouter-x', 'ubnt_edgerouter-x', {
