diff --git a/targets/mediatek-mt7622 b/targets/mediatek-mt7622
index 81440d50..78f0ba5b 100644
--- a/targets/mediatek-mt7622
+++ b/targets/mediatek-mt7622
@@ -13,3 +13,17 @@ device('ubiquiti-unifi-6-lr-v1', 'ubnt_unifi-6-lr-v1', {
 	manifest_aliases = {'ubiquiti-unifi-6-lr'}, -- Upgrade from OpenWrt 22.03
 })
 
+device('ubiquiti-unifi-6-lr-v2', 'ubnt_unifi-6-lr-v2', {
+	factory = false,
+})
+
+device('ubiquiti-unifi-6-lr-v3', 'ubnt_unifi-6-lr-v3', {
+	factory = false,
+})
+
+
+-- Netgear
+
+device('netgear-wax206', 'netgear_wax206', {
+	factory_ext = '.img',
+})
