diff --git a/targets/ramips-mt7621 b/targets/ramips-mt7621
index b183bc58..e70a3574 100644
--- a/targets/ramips-mt7621
+++ b/targets/ramips-mt7621
@@ -7,6 +7,33 @@ device('asus-rt-ac57u', 'asus_rt-ac57u', {
 
 -- Cudy
 
+device('cudy-x6', 'cudy_x6', {
+        factory = false,
+})
+
+device('cudy-wr1300', 'cudy_wr1300', {
+        factory = false,
+})
+
+--device('cudy-m1800', 'cudy_m1800', {
+--       factory = false,
+--})
+--
+--device('cudy-wr1300-v1', 'cudy_wr1300-v1', {
+--       factory = false,
+--})
+--
+--device('cudy-wr1300-v2', 'cudy_wr1300-v2', {
+--       factory = false,
+--       manifest_aliases = {
+--               'cudy-wr1300-v3',
+--       },
+--})
+
+device('cudy-wr2100', 'cudy_wr2100', {
+        factory = false,
+})
+
 device('cudy-wr2100', 'cudy_wr2100', {
 	factory = false,
 })
