diff --git a/targets/targets.mk b/targets/targets.mk
index 3a885f22..52186339 100644
--- a/targets/targets.mk
+++ b/targets/targets.mk
@@ -26,5 +26,8 @@ $(eval $(call GluonTarget,x86,64))
 
 ifneq ($(BROKEN),)
 $(eval $(call GluonTarget,bcm27xx,bcm2710)) # BROKEN: Untested
+$(eval $(call GluonTarget,bcm27xx,bcm2711)) # BROKEN: Untested
+$(eval $(call GluonTarget,ipq40xx,chromium)) # BROKEN: Untested
+$(eval $(call GluonTarget,ipq807x,generic)) # BROKEN: Untested
 $(eval $(call GluonTarget,mvebu,cortexa9)) # BROKEN: No 11s support
 endif
