diff --git a/targets/ath79-generic b/targets/ath79-generic
index aafcb22f..cd697d80 100644
--- a/targets/ath79-generic
+++ b/targets/ath79-generic
@@ -57,6 +57,10 @@ device('avm-fritz-wlan-repeater-450e', 'avm_fritz450e', {
 	factory = false,
 })
 
+device('avm-fritz-wlan-repeater-1750e', 'avm_fritz1750e', {
+	factory = false,
+})
+
 -- Buffalo
 
 device('buffalo-wzr-hp-ag300h', 'buffalo_wzr-hp-ag300h')
@@ -468,6 +472,15 @@ device('tp-link-eap225-outdoor-v1', 'tplink_eap225-outdoor-v1', {
 	packages = ATH10K_PACKAGES_QCA9888,
 })
 
+device('tp-link-eap225-outdoor-v3', 'tplink_eap225-outdoor-v3', {
+	factory = false,
+	packages = ATH10K_PACKAGES_QCA9888,
+})
+
+device('tp-link-eap225-wall-v2', 'tplink_eap225-wall-v2', {
+	packages = ATH10K_PACKAGES_QCA9888,
+})
+
 device('tp-link-tl-wdr3500-v1', 'tplink_tl-wdr3500-v1')
 device('tp-link-tl-wdr3600-v1', 'tplink_tl-wdr3600-v1')
 device('tp-link-tl-wdr4300-v1', 'tplink_tl-wdr4300-v1')
@@ -581,3 +594,14 @@ device('ubiquiti-unifi-ap-outdoor+', 'ubnt_unifi-ap-outdoor-plus', {
 })
 
 device('ubiquiti-unifi-ap-pro', 'ubnt_unifi-ap-pro')
+
+-- device('ubiquiti-rocket-5ac-lite', 'ubnt_rocket-5ac-lite', {
+--	packages = ATH10K_PACKAGES_QCA9880,
+-- })
+
+
+-- Zyxel
+
+device('zyxel-nbg6616', 'zyxel_nbg6616', {
+	packages = ATH10K_PACKAGES_QCA9880,
+})
