diff --git a/targets/ath79-generic b/targets/ath79-generic
index dc884946..658be02d 100644
--- a/targets/ath79-generic
+++ b/targets/ath79-generic
@@ -474,22 +474,22 @@ device('tp-link-eap225-outdoor-v1', 'tplink_eap225-outdoor-v1', {
 	packages = ATH10K_PACKAGES_QCA9888,
 })
 
-device('tp-link-re355-v1', 'tplink_re355-v1', {
-	manifest_aliases = {
-		'tp-link-re355', -- upgrade from OpenWrt 19.07
-	},
-	packages = ATH10K_PACKAGES_SMALLBUFFERS_QCA9880,
-	broken = true, -- OOM with 5GHz enabled in most environments if device is 64M RAM variant
-	class = 'tiny', -- Only 6M of usable Firmware space
-})
-
-device('tp-link-re450-v1', 'tplink_re450-v1', {
-	packages = ATH10K_PACKAGES_QCA9880,
-	manifest_aliases = {
-		'tp-link-re450', -- upgrade from OpenWrt 19.07
-	},
-	class = 'tiny', -- Only 6M of usable Firmware space
-})
+--device('tp-link-re355-v1', 'tplink_re355-v1', {
+--	manifest_aliases = {
+--		'tp-link-re355', -- upgrade from OpenWrt 19.07
+--	},
+--	packages = ATH10K_PACKAGES_SMALLBUFFERS_QCA9880,
+--	broken = true, -- OOM with 5GHz enabled in most environments if device is 64M RAM variant
+--	class = 'tiny', -- Only 6M of usable Firmware space
+--})
+
+--device('tp-link-re450-v1', 'tplink_re450-v1', {
+--	packages = ATH10K_PACKAGES_QCA9880,
+--	manifest_aliases = {
+--		'tp-link-re450', -- upgrade from OpenWrt 19.07
+--	},
+--	class = 'tiny', -- Only 6M of usable Firmware space
+--})
 
 device('tp-link-tl-wdr3500-v1', 'tplink_tl-wdr3500-v1')
 device('tp-link-tl-wdr3600-v1', 'tplink_tl-wdr3600-v1')
@@ -503,11 +503,11 @@ device('tp-link-tl-wr842n-v3', 'tplink_tl-wr842n-v3', {
 	},
 })
 
-device('tp-link-tl-wr902ac-v1', 'tplink_tl-wr902ac-v1', {
-	packages = ATH10K_PACKAGES_SMALLBUFFERS_QCA9887,
-	broken = true, -- OOM with 5GHz enabled in most environments
-	class = 'tiny', -- 64M ath9k + ath10k
-})
+--device('tp-link-tl-wr902ac-v1', 'tplink_tl-wr902ac-v1', {
+--	packages = ATH10K_PACKAGES_SMALLBUFFERS_QCA9887,
+--	broken = true, -- OOM with 5GHz enabled in most environments
+--	class = 'tiny', -- 64M ath9k + ath10k
+--})
 
 device('tp-link-tl-wr1043nd-v2', 'tplink_tl-wr1043nd-v2', {
 	manifest_aliases = {
@@ -600,3 +600,14 @@ device('ubiquiti-unifi-ap-outdoor+', 'ubnt_unifi-ap-outdoor-plus', {
 })
 
 device('ubiquiti-unifi-ap-pro', 'ubnt_unifi-ap-pro')
+
+device('ubiquiti-rocket-5ac-lite', 'ubnt_rocket-5ac-lite', {
+	packages = ATH10K_PACKAGES_QCA9880,
+})
+
+
+-- Zyxel
+
+device('zyxel-nbg6616', 'zyxel_nbg6616', {
+	packages = ATH10K_PACKAGES_QCA9880,
+})
