diff --git a/targets/ramips-mt76x8 b/targets/ramips-mt76x8
index 52fb71be..382c3dfa 100644
--- a/targets/ramips-mt76x8
+++ b/targets/ramips-mt76x8
@@ -62,9 +62,9 @@ device('tp-link-re200-v2', 'tplink_re200-v2')
 
 device('tp-link-re200-v3', 'tplink_re200-v3')
 
-device('tp-link-re305', 'tplink_re305-v1', {
-	class = 'tiny', -- Only 6M of usable Firmware space
-})
+--device('tp-link-re305', 'tplink_re305-v1', {
+--	class = 'tiny', -- Only 6M of usable Firmware space
+--})
 
 device('tp-link-tl-mr3020-v3', 'tplink_tl-mr3020-v3', {
 	factory = false,
