diff --git a/targets/ramips-mt76x8 b/targets/ramips-mt76x8
index b11b7672..95a4c343 100644
--- a/targets/ramips-mt76x8
+++ b/targets/ramips-mt76x8
@@ -80,6 +80,14 @@ device('tp-link-tl-mr3420-v5', 'tplink_tl-mr3420-v5', {
 	},
 })
 
+device('tp-link-tl-mr6400-v5', 'tplink_tl-mr6400-v5', {
+	factory = false,
+	extra_images = {
+		{'-squashfs-tftp-recovery', '-bootloader', '.bin'},
+	},
+})
+
+
 device('tp-link-tl-wa801nd-v5', 'tplink_tl-wa801nd-v5', {
 	factory = false,
 	extra_images = {
