diff --git a/targets/ipq806x-generic b/targets/ipq806x-generic
index 641d89ce..d149750d 100644
--- a/targets/ipq806x-generic
+++ b/targets/ipq806x-generic
@@ -11,6 +11,12 @@ local QCA9984_PACKAGES = {'kmod-ath10k', '-kmod-ath10k-ct', 'ath10k-firmware-qca
 -- QCA9980 / QCA9990
 --
 
+-- Aruba
+
+device('aruba-ap-325', 'aruba_ap-325', {
+	packages = QCA9980_PACKAGES,
+})
+
 -- Extreme Networks
 
 device('extreme-networks-ap3935', 'extreme_ap3935', {
