diff --git a/targets/ramips-mt7620 b/targets/ramips-mt7620
index fbf57ba8..c9fd5627 100644
--- a/targets/ramips-mt7620
+++ b/targets/ramips-mt7620
@@ -31,14 +31,6 @@ device('netgear-ex3700', 'netgear_ex3700', {
 
 -- Nexx
 
-device('nexx-wt3020-8m', 'nexx_wt3020-8m', {
-	aliases = {
-		'nexx-wt3020ad',
-		'nexx-wt3020f',
-		'nexx-wt3020h',
-	},
-})
-
 -- TP-Link
 
 local tplink_region_suffix = ''
