diff --git a/targets/ramips-mt7621 b/targets/ramips-mt7621
index a4d67702..cb699c68 100644
--- a/targets/ramips-mt7621
+++ b/targets/ramips-mt7621
@@ -7,6 +7,10 @@ device('asus-rt-ac57u-v1', 'asus_rt-ac57u-v1', {
 	},
 })
 
+device('asus-rt-ax53u', 'asus_rt-ax53u', {
+	factory = false,
+})
+
 
 -- Cudy
 
@@ -55,6 +59,15 @@ device('gl.inet-gl-mt1300', 'glinet_gl-mt1300', {
 device('mercusys-mr70x-v1', 'mercusys_mr70x-v1')
 
 
+-- MikroTik
+
+device('mikrotik-routerboard-750gr3', 'mikrotik_routerboard-750gr3', {
+	aliases = {'mikrotik-routerboard-hex-v3'},
+	factory = false,
+	packages = {'-hostapd-mini'},
+})
+
+
 -- Netgear
 
 device('netgear-ex6150', 'netgear_ex6150', {
