diff --git a/targets/ath79-generic b/targets/ath79-generic
index 38041833..2bf0dbfb 100644
--- a/targets/ath79-generic
+++ b/targets/ath79-generic
@@ -412,11 +412,11 @@ device('tp-link-archer-c58-v1', 'tplink_archer-c58-v1', {
 --	},
 --})
 
-device('tp-link-archer-c60-v1', 'tplink_archer-c60-v1', {
-	packages = ATH10K_PACKAGES_QCA9888,
-	broken = true, -- OOM with 5GHz enabled in most environments
-	class = 'tiny', -- 64M ath9k + ath10k
-})
+--device('tp-link-archer-c60-v1', 'tplink_archer-c60-v1', {
+--	packages = ATH10K_PACKAGES_QCA9888,
+--	broken = true, -- OOM with 5GHz enabled in most environments
+--	class = 'tiny', -- 64M ath9k + ath10k
+--})
 
 device('tp-link-archer-c7-v2', 'tplink_archer-c7-v2', {
 	packages = ATH10K_PACKAGES_QCA9880,
@@ -503,11 +503,11 @@ device('tp-link-tl-wr842n-v3', 'tplink_tl-wr842n-v3', {
 	},
 })
 
-device('tp-link-tl-wr902ac-v1', 'tplink_tl-wr902ac-v1', {
-	packages = ATH10K_PACKAGES_SMALLBUFFERS_QCA9887,
-	broken = true, -- OOM with 5GHz enabled in most environments
-	class = 'tiny', -- 64M ath9k + ath10k
-})
+--device('tp-link-tl-wr902ac-v1', 'tplink_tl-wr902ac-v1', {
+--	packages = ATH10K_PACKAGES_SMALLBUFFERS_QCA9887,
+--	broken = true, -- OOM with 5GHz enabled in most environments
+--	class = 'tiny', -- 64M ath9k + ath10k
+--})
 
 device('tp-link-tl-wr1043nd-v2', 'tplink_tl-wr1043nd-v2', {
 	manifest_aliases = {
@@ -600,3 +600,14 @@ device('ubiquiti-unifi-ap-outdoor+', 'ubnt_unifi-ap-outdoor-plus', {
 })
 
 device('ubiquiti-unifi-ap-pro', 'ubnt_unifi-ap-pro')
+
+device('ubiquiti-rocket-5ac-lite', 'ubnt_rocket-5ac-lite', {
+	packages = ATH10K_PACKAGES_QCA9880,
+})
+
+
+-- Zyxel
+
+device('zyxel-nbg6616', 'zyxel_nbg6616', {
+	packages = ATH10K_PACKAGES_QCA9880,
+})
