diff --git a/targets/mediatek-filogic b/targets/mediatek-filogic
index ce398bda..217256e5 100644
--- a/targets/mediatek-filogic
+++ b/targets/mediatek-filogic
@@ -51,6 +58,24 @@ device('cudy-wr3000e-v1', 'cudy_wr3000e-v1', {
 	factory = false,
 })
 
+device('cudy-wr3000s-v1', 'cudy_wr3000s-v1', {
+        factory = false,
+})
+
+device('cudy-re3000-v1', 'cudy_re3000-v1', {
+        factory = false,
+})
+
+device('cudy-wr3000h-v1', 'cudy_wr3000h-v1', {
+        factory = false,
+})
+
+-- Mercusys
+
+device('mercusys-mr90x-v1', 'mercusys_mr90x-v1', {
+	factory = false,
+})
+
 -- GL.iNet
 
 device('gl.inet-gl-mt2500', 'glinet_gl-mt2500', {
