diff --git a/targets/ath79-nand b/targets/ath79-nand
index ddbbc7c9..6e654cc4 100644
--- a/targets/ath79-nand
+++ b/targets/ath79-nand
@@ -1,3 +1,11 @@
+local ATH10K_PACKAGES_QCA9880 = {
+	'kmod-ath10k',
+	'-kmod-ath10k-ct',
+	'-kmod-ath10k-ct-smallbuffers',
+	'ath10k-firmware-qca988x',
+	'-ath10k-firmware-qca988x-ct',
+}
+
 local ATH10K_PACKAGES_QCA9887 = {
 	'kmod-ath10k',
 	'-kmod-ath10k-ct',
@@ -52,3 +60,10 @@ device('zte-mf281', 'zte_mf281', {
 	broken = true,
 	packages = ATH10K_PACKAGES_QCA9888,
 })
+
+-- Zyxel
+
+device('zyxel-nbg6716', 'zyxel_nbg6716', {
+	packages = ATH10K_PACKAGES_QCA9880,
+	sysupgrade_ext = '.tar',
+})
