diff --git a/targets/ipq40xx-generic b/targets/ipq40xx-generic
index 246fe066..1247cac4 100644
--- a/targets/ipq40xx-generic
+++ b/targets/ipq40xx-generic
@@ -87,6 +87,21 @@ device('gl.inet-gl-b1300', 'glinet_gl-b1300', {
 
 device('linksys-ea6350v3', 'linksys_ea6350v3')
 
+device('linksys-mr8300-dallas', 'linksys_mr8300', {
+	packages = { -- exclude until Gluon supports third radio
+		'-ath10k-firmware-qca9888-ct',
+		'-ath10k-board-qca9888',
+	},
+	factory = false,
+	broken = true, -- 3rd WiFi radio not supported (5GHz DFS channels 100+)
+})
+
+-- Meraki
+
+device('meraki-mr33', 'meraki_mr33', {
+        factory = false,
+})
+
 
 -- NETGEAR
 
