diff --git a/targets/targets.mk b/targets/targets.mk
index a2eec7dc..71047d56 100644
--- a/targets/targets.mk
+++ b/targets/targets.mk
@@ -30,6 +30,7 @@ ifeq ($(BROKEN),1)
 $(eval $(call GluonTarget,bcm27xx,bcm2710)) # BROKEN: Untested
 $(eval $(call GluonTarget,bcm27xx,bcm2711)) # BROKEN: No 11s support, no reset button, sys LED issues
 $(eval $(call GluonTarget,ipq40xx,chromium)) # BROKEN: Devices cannot be flashed without opening the case
+$(eval $(call GluonTarget,qualcommax,ipq807x)) # BROKEN: Untested
 $(eval $(call GluonTarget,kirkwood,generic)) # BROKEN: No devices with 11s support
 $(eval $(call GluonTarget,mvebu,cortexa9)) # BROKEN: No 11s support
 endif
