diff --git a/targets/ipq40xx-generic b/targets/ipq40xx-generic
index 83b4a128..c5120cbe 100644
--- a/targets/ipq40xx-generic
+++ b/targets/ipq40xx-generic
@@ -96,6 +96,13 @@ device('gl.inet-gl-b1300', 'glinet_gl-b1300', {
 device('linksys-ea6350v3', 'linksys_ea6350v3')
 
 
+-- Meraki
+
+device('meraki-mr33', 'meraki_mr33', {
+	factory = false,
+})
+
+
 -- NETGEAR
 
 device('netgear-ex6100v2', 'netgear_ex6100v2', {
