diff --git a/targets/qualcommax-ipq807x b/targets/qualcommax-ipq807x
new file mode 100644
index 00000000..5d70778b
--- /dev/null
+++ b/targets/qualcommax-ipq807x
@@ -0,0 +1,3 @@
+device('xiaomi-ax3600', 'xiaomi_ax3600', {
+        broken = true,
+})
diff --git a/targets/targets.mk b/targets/targets.mk
index 77940da5..9afa5510 100644
--- a/targets/targets.mk
+++ b/targets/targets.mk
@@ -28,4 +28,5 @@ ifneq ($(BROKEN),)
 $(eval $(call GluonTarget,bcm27xx,bcm2710)) # BROKEN: Untested
 $(eval $(call GluonTarget,bcm27xx,bcm2711)) # BROKEN: Untested
 $(eval $(call GluonTarget,mvebu,cortexa9)) # BROKEN: No 11s support
+$(eval $(call GluonTarget,qualcommax,ipq807x)) # BROKEN: Untested
 endif
