diff --git a/targets/lantiq-xrx200 b/targets/lantiq-xrx200
index 364c0c31..7590efd5 100644
--- a/targets/lantiq-xrx200
+++ b/targets/lantiq-xrx200
@@ -47,6 +47,10 @@ device('avm-fritz-box-7412', 'avm_fritz7412', {
 	factory = false,
 })
 
+device('avm-fritz-box-7430', 'avm_fritz7430', {
+	factory = false,
+})
+
 -- TP-Link
 
   -- CAVEAT: These devices don't have a dedicated WAN port.
@@ -63,3 +67,11 @@ device('tp-link-td-w8980', 'tplink_tdw8980', {
 	aliases = {'tp-link-td-w9980'},
 	broken = true, -- 5GHz unsupported
 })
+
+
+-- O2
+
+device('arcadyan-vgv7510kw22', 'arcadyan_vgv7510kw22-nor', {
+	factory = false,
+	aliases = {'o2-box-6431'},
+})

\ No newline at end of file
