diff --git a/targets/ramips-mt7620 b/targets/ramips-mt7620
index fbf57ba8..ed06eee2 100644
--- a/targets/ramips-mt7620
+++ b/targets/ramips-mt7620
@@ -29,16 +29,6 @@ device('netgear-ex3700', 'netgear_ex3700', {
 	factory_ext = '.chk',
 })
 
--- Nexx
-
-device('nexx-wt3020-8m', 'nexx_wt3020-8m', {
-	aliases = {
-		'nexx-wt3020ad',
-		'nexx-wt3020f',
-		'nexx-wt3020h',
-	},
-})
-
 -- TP-Link
 
 local tplink_region_suffix = ''
@@ -46,20 +36,20 @@ if (env.GLUON_REGION or '') ~= '' then
 	tplink_region_suffix = '-' .. env.GLUON_REGION
 end
 
-device('tp-link-archer-c2-v1', 'tplink_archer-c2-v1', {
-	factory = false,
-})
+--device('tp-link-archer-c2-v1', 'tplink_archer-c2-v1', {
+--	factory = false,
+--})
 
-device('tp-link-archer-c20-v1', 'tplink_archer-c20-v1', {
-	factory = false,
-})
+--device('tp-link-archer-c20-v1', 'tplink_archer-c20-v1', {
+--	factory = false,
+--})
 
-device('tp-link-archer-c20i', 'tplink_archer-c20i')
+--device('tp-link-archer-c20i', 'tplink_archer-c20i')
 
-device('tp-link-archer-c50-v1', 'tplink_archer-c50-v1', {
-	factory = '-squashfs-factory' .. tplink_region_suffix,
-	manifest_aliases = {'tp-link-archer-c50'},
-})
+--device('tp-link-archer-c50-v1', 'tplink_archer-c50-v1', {
+--	factory = '-squashfs-factory' .. tplink_region_suffix,
+--	manifest_aliases = {'tp-link-archer-c50'},
+--})
 
 
 -- Xiaomi
