diff --git a/targets/targets.mk b/targets/targets.mk
index 14955134..7a5a9b2e 100644
--- a/targets/targets.mk
+++ b/targets/targets.mk
@@ -29,5 +29,7 @@ $(eval $(call GluonTarget,x86,64))
 ifeq ($(BROKEN),1)
 $(eval $(call GluonTarget,bcm27xx,bcm2710)) # BROKEN: Untested
 $(eval $(call GluonTarget,bcm27xx,bcm2711)) # BROKEN: No 11s support, no reset button, sys LED issues
+$(eval $(call GluonTarget,ipq40xx,chromium)) # BROKEN: Untested
+$(eval $(call GluonTarget,ipq807x,generic)) # BROKEN: Untested
 $(eval $(call GluonTarget,mvebu,cortexa9)) # BROKEN: No 11s support
 endif
